webui/react/src/pages/WorkspaceDetails/WorkspaceMembers.tsx

```diff
@@ -0,0 +1,266 @@
+  assignments,
+  usersAssignedDirectly,
+  groupsAssignedDirectly,
+  pageRef,
+  workspace,
+}: Props) => {
+  const { knownRoles } = useStore();
+
+  const { canUpdateRoles } = usePermissions();
+
+  const { settings, updateSettings } = useSettings<WorkspaceMembersSettings>(settingsConfig);
+  const userCanAssignRoles = canUpdateRoles({ workspace });
+
+  const usersAndGroups: UserOrGroup[] = useMemo(
+    () => [...usersAssignedDirectly, ...groupsAssignedDirectly],
+    [groupsAssignedDirectly, usersAssignedDirectly],
+  );
+
+  const handleNameSearchApply = useCallback(
+    (newSearch: string) => {
+      updateSettings({ name: newSearch || undefined });
+    },
+    [updateSettings],
+  );
+
+  const handleNameSearchReset = useCallback(() => {
+    updateSettings({ name: undefined });
+  }, [updateSettings]);
+
+  const nameFilterSearch = useCallback(
+    (filterProps: FilterDropdownProps) => (
+      <TableFilterSearch
+        {...filterProps}
+        value={settings.name || ''}
+        onReset={handleNameSearchReset}
+        onSearch={handleNameSearchApply}
+      />
+    ),
+    [handleNameSearchApply, handleNameSearchReset, settings.name],
+  );
+
+  const tableSearchIcon = useCallback(() => <Icon name="search" size="tiny" />, []);
+
+  const columns = useMemo(() => {
+    const nameRenderer = (value: string, record: UserOrGroup) => {
+      if (isUser(record)) {
+        const member = record as User;
+        return (
+          <>
+            <div className={css.userAvatarRowItem}>
+              <Avatar size={Size.Medium} userId={member.id} />
+            </div>
+            <div className={css.userRowItem}>
+              {member?.displayName ? (
+                <>
+                  <div>{member.displayName}</div>
+                  <div>{member.username}</div>
+                </>
+              ) : (
+                <div>{member.username}</div>
+              )}
+            </div>
+          </>
+        );
+      }
+      const group = record as V1GroupDetails;
+      return (
+        <>
+          <div className={css.userAvatarRowItem}>
+            <Icon name="group" />
+          </div>
+          <div className={css.userRowItem}>
+            <div>{group.name}</div>
+          </div>
+        </>
+      );
+    };
+
+    const roleRenderer = (value: string, record: UserOrGroup) => {
+      return (
+        <Select
+          className={css.selectContainer}
+          disabled={!userCanAssignRoles}
+          value={assignments[0]}
+          onSelect={async (value: RawValueType | LabelInValueType) => {
+            const roleIdValue = value as number;
+            const userOrGroupId = getIdFromUserOrGroup(record);
+
+            // Needs to be updated to get the correct old role for the user
+            // or group.
+            const oldRoleId = assignments?.[0].role?.roleId || 0;
+
+            // Remove the old role then add the new role
+            if (isUser(record)) {
```

Comment:
Wanted to confirm the logic here is correct? For the dropdown in the table in https://www.figma.com/file/kqWThAhkfa4giVOwHzrAxA/RBAC-Web?node-id=39%3A5166

We will need to remove the old `role` and add the new `role` correct? There is not a single API call to do so?